### PR TITLE
MySQL cypress test

### PR DESCRIPTION
**Pull request by Opalevanescence**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Adds a MySQL folder to the metabase-db folder
- Checks you can add a MySQL database to your Metabase instance 
- Checks you can ask a simple question
- Checks you can ask a Native query
- Checks you can save a Native query

### How to test this manually?
- `docker run --rm -p 3306:3306 --name meta-mysql8-sample metabase/qa-databases:mysql-sample-8`
- `yarn test-cypress-open --testFiles frontend/test/metabase-db/mysql` and run all tests

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
@Opalevanescence Which command did you use to spin up that `mysql-sample-8` Docker image? I would like to try it locally. You can include that in the description under "How to test this manually?". Thanks.

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]
Please "uncommit" `locales.clj` file.

1. That file changes locally when we run `./bin/build no-translations` as a workaround for Macs, but it shouldn't be committed.
2. In general - even if some other file (in some imaginary scenario) **should** be committed but is **not** related to the PR directly, I'd use a separate PR just for that file. It makes PRs cleaner, focused on a single feature and easier to review.

File: frontend/test/metabase-db/mysql/query.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    restore();
+    signInAsAdmin();
+    addMySqlDatabase();
+  });
+
+  beforeEach(() => {
+    signInAsNormalUser();
+  });
+
+  it("can query a My SQL database as a user", () => {
```
Comment (nemanjaglumac) [inline frontend/test/metabase-db/mysql/query.cy.spec.js:53]:
```suggestion
  it("can query a MySQL database as a user", () => {
```

File: frontend/test/metabase-db/mysql/query.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    cy.visit("/question/new");
+    cy.contains("Simple question").click();
+    cy.contains("MySQL").click();
+    cy.contains("Orders").click();
+    cy.contains("Showing first 2,000 rows");
```
Comment (nemanjaglumac) [inline frontend/test/metabase-db/mysql/query.cy.spec.js:58]:
```suggestion
```

I'd advise against using number of rows displayed in assertions. I've run into a couple of problems with some older tests where that assertion had to be removed (number of rows changes based on the screen size and is fragile as such).

File: frontend/test/metabase-db/mysql/query.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    cy.visit("/question/new");
+    cy.contains("Native query").click();
+    cy.contains("MySQL").click();
+
+    cy.get(".ace_content").type(`SELECT * FROM ORDERS`, {
+      parseSpecialCharSequences: false,
+    });
+    cy.get(".NativeQueryEditor .Icon-play").click();
+    cy.contains("2,000");
```
Comment (nemanjaglumac) [inline frontend/test/metabase-db/mysql/query.cy.spec.js:97]:
```suggestion
    cy.contains("37.65");
```

The same comment/reason against using number of rows as above.

File: frontend/test/metabase-db/mysql/query.cy.spec.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    // Filter by Doohickey
+    cy.findByPlaceholderText("Category")
+      .click()
+      .type("Doohickey");
+    cy.get(".NativeQueryEditor .Icon-play").click();
+    cy.contains("Widget").should("not.exist");
+    cy.contains("Doohickey");
+  });
+
+  it("can save a native My SQL query", () => {
```
Comment (nemanjaglumac) [inline frontend/test/metabase-db/mysql/query.cy.spec.js:85]:
```suggestion
  it("can save a native MySQL query", () => {
```

File: frontend/test/metabase-db/mysql/query.cy.spec.js (hunk 2 above)
Comment (Opalevanescence) [reply frontend/test/metabase-db/mysql/query.cy.spec.js:58]:
```suggestion
    cy.contains("Showing first");
```

File: resources/locales.clj [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  :locales  #{"en" "nl" "zh" "zh-HK" "fr" "ja" "bg" "de" "nb" "es" "cs" "ca" "pl" "zh-TW" "vi" "it" "ru" "sv" "fa" "sk" "pt" "tr" }
+  :locales  #{"en" "zh" "cs" "fr" "nl" "sv" "tr" "sk" "it" "es" "nb" "ca" "de" "fa" "zh-TW" "bg" "ja" "zh-HK" "vi" "pl" "pt" "ru" }
```
Comment (Opalevanescence) [inline resources/locales.clj:2]:
```suggestion
  :locales  #{"en" "nl" "zh" "zh-HK" "fr" "ja" "bg" "de" "nb" "es" "cs" "ca" "pl" "zh-TW" "vi" "it" "ru" "sv" "fa" "sk" "pt" "tr" }
```

File: frontend/test/metabase-db/mysql/add.cy.spec.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    // We're clicking on ID: 1 (the first order) => do not change!
+    // It is tightly coupled to the assertion ("37.65"), which is "Subtotal" value for that order.
+    cy.get(".Table-ID")
+      .eq(0)
+      .click();
+
+    cy.get(".LoadingSpinner").should("not.exist");
+
+    // Assertions
+    cy.log("**Fails in v0.36.6**");
```
Comment (nemanjaglumac) [inline frontend/test/metabase-db/mysql/add.cy.spec.js:94]:
```suggestion
```

Sorry, I missed this one. That was specific to a PostgreSQL issue and we don't want/need to include that log here.

**Review by nemanjaglumac — APPROVED** [review]
LGTM! 🎉

**Pull request merged**